app/services/media_storage/azure_adapter.rb

```diff
@@ -0,0 +1,85 @@
+module MediaStorage
+  class AzureAdapter < AbstractAdapter
+    attr_reader :prefix, :container, :storage_account_name, :get_expiration, :put_expiration, :client, :signer
+    DEFAULT_EXPIRES_IN = 3 # time in minutes, see get_expiry_time(expires_in)
+
+    def initialize(opts={})
+      @storage_account_name = opts[:azure_storage_account]
+      @container = opts[:azure_storage_container] || Rails.env
+      @prefix = opts[:azure_prefix]
+      @get_expiration = opts.dig(:expiration, :get) || DEFAULT_EXPIRES_IN
+      @put_expiration = opts.dig(:expiration, :put) || DEFAULT_EXPIRES_IN
+
+      @client = Azure::Storage::Blob::BlobService.create(
+        storage_account_name: opts[:azure_storage_account],
+        storage_access_key: opts[:azure_storage_access_key]
+      )
+      @signer = Azure::Storage::Common::Core::Auth::SharedAccessSignature.new(
+        opts[:azure_storage_account],
+        opts[:azure_storage_access_key]
+      )
+    end
+
+    def stored_path(content_type, medium_type, *path_prefix)
+      extension = get_extension(content_type)
+      path = prefix.to_s
+      path += '/' unless path[-1] == '/'
+      path += container + '/'
```

Comment:
hmm - why are we including the container in the stored path? 

my understanding is we store the object in the container at the stored path. The container should not be included in the path.

Reply:
Yes you are right. I was thinking "stored path" more along the lines of "what is the URL at which this object can be accessed?" 

Now that Ive looked through how `stored_path` is used to generate the `src` attribute in the Media records, which then later gets passed to the `path` param of the `put_path` or `get_path` methods I see that we dont need the container in the path here.